lib/deaf_grandma.rb

```diff
@@ -17,11 +17,21 @@ def run!
     end
   end
 
-
+  
   def speak(input)
-
     #Implement your code here <<<<<<<<<
-
+    if input == input.upcase
+      if input == "BYE"
+        @bye_counter += 1
+        if @bye_counter == 3
+          puts "SEE YOU LATER SONNY!"
+          exit
+        end
+      end
+       "NOT SINCE 1964!"
```

Comment:
I see how this works: you'll only get to 'not since 1964' if you haven't exited in the previous if-clause.  Still, an `else` might improve readability.